### ISSUE TITLE
Publish as web extension

### ISSUE THREAD

**Issue by anderseknert** [open-policy-agent/vscode-opa#139]
Web extensions are extensions that can be installed in the web interface for VS Code, such as presented in GitHub et. al. 

While we can't do things like calling out to the OPA binary in this context, we could still provide basic features like syntax highlighting.

Note that extensions can be published both as a "desktop" version and a web one, which is what we'll want to do here. We don't want to limit the capabilities of the desktop one, so this would be a new additional part of the project.

More info here: https://code.visualstudio.com/api/extension-guides/web-extensions

(It's even possible to run a language server inside of a web worker, so we _could_ compile Regal to Wasm (as @srenatus previously demonstrated) or if not that, at least the Regal policies, and have linting enabled too. But let's start with the basics)

**Comment by srenatus** [comment]
> While we can't do things like calling out to the OPA binary in this context, we could still provide basic features like syntax highlighting.

It has to be tried, but if we can compile regal to Wasm _and use it_, then it should work for `opa eval`, too.

**Comment by anderseknert** [comment]
That's true, but things like evaluation isn't covered by a language server, at least not without custom extensions. I guess it's possible to run parts of the _client_ in Wasm too, and if so we could have OPA run there. Considering the size and performance of Regal compiled to Wasm last time we tried, perhaps a good middle ground, or at least first step, would be to compile the Regal policies only, and invoke those inside of the LSP web worker 🤔 Oh well, things may look very different when we get there.

**Comment by nikpivkin** [comment]
Hi @anderseknert !

TCP [support](https://github.com/StyraInc/regal/issues/764) on the LSP server side won't solve the problem with connecting extensions from the Web?

**Comment by charlieegan3** [comment]
It'd be part of it for sure, the other major parts to this as I understand it are:

- something to read in the file contents (rather than just the open buffers), the local version of the server reads from disk.
- some system of hosting a TCP Regal LS instance that supports multiple users. Even if this were to be a self hosted component, we'd need to think of a good way to make this available for users (container image etc.).

**Comment by nikpivkin** [comment]
I think I accidentally deleted the comment

TCP [support](https://github.com/StyraInc/regal/issues/764) on the LSP server side won't solve the problem with connecting extensions from the Web?
